samples/otlptrace/example_grpc.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+import google.auth
+import google.auth.transport.grpc
+import google.auth.transport.requests
+import grpc
+from google.auth.transport.grpc import AuthMetadataPlugin
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import (
+    OTLPSpanExporter,
+)
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+"""
+This is a sample script that exports OTLP traces encoded as protobufs via gRPC. 
+"""
+
+
+credentials, project_id = google.auth.default()
+request = google.auth.transport.requests.Request()
+credentials.refresh(request)
+resource = Resource.create(attributes={SERVICE_NAME: "otlp-gcp-grpc-sample"})
+
+auth_metadata_plugin = AuthMetadataPlugin(
+    credentials=credentials, request=request
+)
+channel_creds = grpc.composite_channel_credentials(
+    grpc.ssl_channel_credentials(),
+    grpc.metadata_call_credentials(auth_metadata_plugin),
+)
+
+trace_provider = TracerProvider(resource=resource)
+processor = BatchSpanProcessor(
+    OTLPSpanExporter(credentials=channel_creds, insecure=False)
```

Comment:
Based on the control flow, in `OTLPExporterMixin`: 

```py
 if insecure:
            self._client = self._stub(
                insecure_channel(self._endpoint, compression=compression)
            )
        else:
            credentials = _get_credentials(
                credentials, OTEL_EXPORTER_OTLP_CERTIFICATE
            )
            self._client = self._stub(
                secure_channel(
                    self._endpoint, credentials, compression=compression
                )
            )
```

Makes me think that in an insecure channel, credentials would not be passed. However, I did not try running the example with `insecure=True`

Reply:
Gotcha. But can we instead rely on the default behavior and just remove the argument altogether?

Reply:
I see - it seems like the default is `None` but if the parsed url scheme is https, it is automatically assigned false.

This makes me think that it depends on the endpoint. If the endpoint uses `http` instead of `https`, then without explicitly defining `insecure=False`, the code might fail.

Does this make sense ?

Reply:
Yes I think it should behave correctly